File: backend/crawl/dhnews.js
const axios = require('axios');
const cheerio = require('cheerio');

async function crawlDhNews() {
  try {
    const url = 'https://dhnews.co.kr/news/cate/';

    // 웹페이지 요청
    const { data } = await axios.get(url);

    // HTML 파싱
    const $ = cheerio.load(data);
    const newsList = $('div#listWrap div.listPhoto');

    const result = [];

    newsList.each((_, el) => {
      const titleTag = $(el).find('dl dt a');
      const dateTag = $(el).find('dd.winfo span.date');

      if (titleTag.length && dateTag.length) {
        const title = titleTag.text().trim();
        const date = dateTag.text().trim();
        const link = 'https://dhnews.co.kr' + titleTag.attr('href');

        result.push({ title, link, date });
      }
    });

    console.log({ "대학저널": result });
    return result;
  } catch (error) {
    console.error('크롤링 실패:', error.message);
    return [];
  }
}

// 약 1초
// crawlDhNews()
module.exports = crawlDhNews;

File: backend/crawl/dhnews.py
import requests
from bs4 import BeautifulSoup

def get_data():
    # 가져올 웹사이트 주소
    url = "https://dhnews.co.kr/news/cate/"

    # 웹페이지 요청
    response = requests.get(url)

    # 요청 성공 여부 확인
    if response.status_code == 200:
        # HTML 파싱
        soup = BeautifulSoup(response.text, 'html.parser')

        # 뉴스 데이터 div 선택
        news_list = soup.select('div#listWrap div.listPhoto')

        result = []
        for news in news_list:
            title_tag = news.select_one('dl dt a')
            title = title_tag.text.strip()
            date = news.select_one('dd.winfo span.date').text.strip()
            link = "https://dhnews.co.kr" + title_tag.attrs['href']

            #print(f"제목: {title}")
            #print(f"날짜: {date}")
            #print(f"링크: {link}")
            #print("-" * 100)
            dict_data = {"title": title, "link": link, "date": date}
            result.append(dict_data)

        return {"대학저널": result}

    # 웹사이트 요청 실패 시 출력
    else:
        print(f"Failed to fetch the page, status code: {response.status_code}")
        return {"Error": response.status_code}